src/Castle.Core/DynamicProxy/Generators/BaseClassProxyGenerator.cs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2004-2021 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.DynamicProxy.Generators
+{
+	using System;
+	using System.Collections.Generic;
+	using System.Linq;
+	using System.Reflection;
+
+	using Castle.DynamicProxy.Contributors;
+	using Castle.DynamicProxy.Generators.Emitters.SimpleAST;
+	using Castle.DynamicProxy.Internal;
+
+	internal abstract class BaseClassProxyGenerator : BaseProxyGenerator
```

Comment:
I'd have probably suggested `ClassProxyGeneratorBase` as this class sounds a bit like something else at the moment, also means it'll sort next to class proxy generators.

Reply:
Having a single naming theme for abstract base classes instead of many (`AbstractX`, `BaseX`) is another refactor we should probably do, and I also tend towards `XBase`. Here, I just didn't want to introduce a third name form.

Reply:
Fair enough. I too usually go for `XBase`. `AbstractX` would come from the early days when most people were doing Java and building Castle.